projects/politie/ngx-sherlock/src/lib/derivable-query-list.spec.ts

```diff
@@ -0,0 +1,77 @@
+import { Component, ElementRef, QueryList, ViewChildren } from '@angular/core';
+import { ComponentFixture, TestBed } from '@angular/core/testing';
+import { derivableQueryList } from './derivable-query-list';
+
+describe('DerivableQueryList', () => {
+
+    @Component({
+        template: `
+            <p #item *ngIf="first">First</p>
+            <p #item *ngIf="second">Second</p>
+            <p #item *ngIf="third">Third</p>
+        `,
+    })
+    class TestComponent {
+        first = true;
+        second = false;
+        third = false;
+
+        @ViewChildren('item')
+        private readonly paragraphs!: QueryList<ElementRef<HTMLParagraphElement>>;
+        readonly paragraphs$ = derivableQueryList(() => this.paragraphs);
+    }
+
+    beforeEach(() => TestBed.configureTestingModule({ declarations: [TestComponent] }).compileComponents());
+
+    let fixture: ComponentFixture<TestComponent>;
+    let componentInstance: TestComponent;
+
+    beforeEach(() => {
+        fixture = TestBed.createComponent(TestComponent);
+        ({ componentInstance } = fixture);
+        fixture.detectChanges();
+    });
+
+    it('resolve after initialized', () => {
+        expect(componentInstance.paragraphs$.resolved).toBeFalsy();
+        componentInstance.paragraphs$.markInitialized();
+        componentInstance.paragraphs$.markInitialized();
```

Comment:
This is a separate test, that tests if `markInitialized` won't throw if called twice